src/abilities/Golden-Wyrm.js

```diff
@@ -233,78 +233,131 @@ export default (G) => {
+		 * - If the target unit is missing less than the max heal amount (50), only
+		 *   the missing amount will be transferred.
+		 * - Cannot target full health units.
+		 * - Requires at least 51 remaining health to use.
+		 *
+		 * When upgraded each use buffs Golden Wyrm with 10 regrowth points.
+		 */
 		{
-			//	Type : Can be "onQuery", "onStartPhase", "onDamage"
 			trigger: 'onQuery',
 
-			damages: {
-				pierce: 15,
-				slash: 10,
-				crush: 5,
-			},
-			_targetTeam: Team.enemy,
+			_targetTeam: Team.ally,
+
+			_maxTransferAmount: 50,
 
-			// 	require() :
 			require: function () {
 				if (!this.testRequirements()) {
 					return false;
 				}
 
-				let map = G.grid.getHexMap(
-					this.creature.x - 2,
-					this.creature.y - 2,
-					0,
-					false,
-					matrices.frontnback2hex,
-				);
-				// At least one target
+				/* Golden Wyrm needs to be left with at least 1 hp after the ability. We
+				can't use ability.costs because that is a fixed value, whereas this ability
+				has a dynamic health cost. */
+				if (this.creature.health < this._maxTransferAmount + 1) {
+					this.message = 'Not enough health';
+					return false;
+				}
+
 				if (
-					!this.atLeastOneTarget(map, {
+					!this.atLeastOneTarget(this._getHexes(), {
 						team: this._targetTeam,
+						optTest: this._confirmTarget,
 					})
 				) {
 					return false;
 				}
+
 				return true;
 			},
 
-			// 	query() :
 			query: function () {
-				let ability = this;
-				let wyrm = this.creature;
+				const ability = this;
+				const wyrm = this.creature;
 
 				G.grid.queryCreature({
 					fnOnConfirm: function () {
 						ability.animation(...arguments);
 					},
+					optTest: this._confirmTarget,
 					team: this._targetTeam,
 					id: wyrm.id,
 					flipped: wyrm.flipped,
-					hexes: G.grid.getHexMap(wyrm.x - 2, wyrm.y - 2, 0, false, matrices.frontnback2hex),
+					hexes: this._getHexes(),
 				});
 			},
 
-			//	activate() :
 			activate: function (target) {
-				let ability = this;
-				ability.end();
+				this.end();
 
-				let damage = new Damage(
-					ability.creature, // Attacker
-					ability.damages, // Damage Type
-					1, // Area
-					[], // Effects
-					G,
+				/* The amount of health transferred is the creature's missing life, capped 
+				to 50. */
+				const transferAmount = Math.min(
+					target.stats.health - target.health,
+					this._maxTransferAmount,
 				);
-				target.takeDamage(damage);
 
-				//remove frogger bonus if its found
-				ability.creature.effects.forEach(function (item) {
-					if (item.name == 'Offense++') {
-						item.deleteEffect();
-					}
-				});
+				target.heal(transferAmount);
+
+				/* Damage Golden Wyrm using `.heal()` instead of `.takeDamage()` to apply 
+				raw damage that bypasses dodge, shielded, etc and does not trigger further 
+				effects. */
+				this.creature.heal(-transferAmount);
+
+				// TODO: Should we show messages/hints for this transfer beyond health lost/gained?
```

Comment:
@DreadKnight Should there be some log message or hint beyond the loss/gain of health?

![CleanShot 2021-12-14 at 22 16 33](https://user-images.githubusercontent.com/199204/145997474-cdc7fd90-e040-4ec2-8d9b-932b5f1820af.gif)

Reply:
The tooltips are fine I'm also pointing out ideal log in another comment in a bit.